File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/api/ValorantStoreController.java
package fr.huiitre.tools.modules.riot.valorant.api;

import fr.huiitre.tools.modules.core.common.api.RequiredRole;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.usecase.GetValorantStoreUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.view.ValorantStoreView;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/riot/valorant")
public class ValorantStoreController {

    private final GetValorantStoreUseCase getValorantStoreUseCase;

    public ValorantStoreController(GetValorantStoreUseCase getValorantStoreUseCase) {
        this.getValorantStoreUseCase = getValorantStoreUseCase;
    }

    @RequiredRole(RoleCode.USER)
    @GetMapping("/store")
    public ValorantStoreView getStore(@RequestHeader(value = "X-Riot-Token", required = false) String accessToken,
                                      @RequestParam(required = false) String region) {
        return getValorantStoreUseCase.execute(accessToken, region);
    }
}

File: api/Modules/Riot/Valorant/Api/ValorantStoreController.cs
using Microsoft.AspNetCore.Mvc;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Usecases;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Views;

namespace Tools.Api.Modules.Riot.Valorant.Api;

// Boutique du jour, lue chez Riot. Deux modes : un compte lié (accountId, l'API retrouve ses
// jetons) ou un jeton fourni à la volée par l'appelant (X-Riot-Token + region).
[ApiController]
[Route("riot/valorant")]
public class ValorantStoreController : ControllerBase
{
    [HttpGet("store")]
    public Task<ValorantStoreView> GetStore(
        [FromServices] GetValorantStoreUseCase getValorantStoreUseCase,
        [FromQuery] long? accountId,
        [FromHeader(Name = "X-Riot-Token")] string? accessToken,
        [FromQuery] string? region
    )
    {
        return getValorantStoreUseCase.Execute(accountId, accessToken, region);
    }
}
